Fullscreen Patch + Bugfix

ntninja opened the pull request:
This patch fixes a bug that breaks the browser layout when the extension is being disabled without a restart. Also - and this is the important part - it adds the abilitity to hide the tabbar in fullscreen
mode.

naturallymitchell commented:
Fullscreen is a great improvement, and I'm happy to have be using it.  Thank you alexander255.

After about a day of use, I haven't experienced any bugs, though the 'Firefox menu' isn't always leaving the bottom of the vertical tab bar in all windows.  I think https://addons.mozilla.org/en-US/firefox/addon/firefox-4-ui-fixer/ might be helpful for this problem and other bottom bar cleanup.

Applying the patch was only mildly difficult with the diffing, unzipping, and re-zipping.  install.rdf had some rejections, so I kept the original.

Luckz commented:
With this patch, does actually moving tabs up/down work again in recent Firefox versions?

ntninja commented:
I don't know about your version but it works for me... I can however remember fixing some issues that caused "Vertical Tabs" not to reload correctly.

Looking at the committed source, the change from

```
tabs.addEventListener('dragover', this.onDragOver, false);
```

to

```
this.tabs.addEventListener('dragover', this.onDragOver, false);
```

seems to confirm this, so please check :-) It's very likely!

Yours,
Alexander

ntninja commented:
In order to try it: Just download my branch as a zip-file, rename it to .xpi and open in Firefox

philikon commented:
Thanks for your patch! Sorry for taking so long to respond. Unfortunately, when going fullscreen (e.g. to watch a video), the tab bar is still there for me.

reuben committed:
Get OS X theme /mostly/ working on Australis

reuben committed:
Replace tabs with spaces

vvuk committed:
Merge pull request #3 from reubenmorais/master

Fix OS X theme on Australis. Pinned tabs still weird.

vvuk committed:
remove unnecessary windows right margin, once drawInTitleBar is turned off

darrinhenein committed:
Initial pass at dark/australis UI

darrinhenein committed:
Fixed squished favicons in list osx

darrinhenein committed:
Added xpi build

ncalexan committed:
Replace tooltiptext with updated dynamic tooltip.

Bug 979479 added dynamic tooltips, including keyboard shortcuts, to
various buttons; this change needs to be reflected in the overridden
XUL.  Before updating, this manifested itself as tabs overlaid on each
other, and the error message "undefined entity" (referring to the old
tooltiptext).

See http://hg.mozilla.org/mozilla-central/rev/4dc06a48ed13.

darrinhenein committed:
Merge pull request #1 from ncalexan/update-after-bug-979479

Thank you!

* Replace tooltiptext with updated dynamic tooltip.

darrinhenein committed:
New xpi

dhdemerson committed:
Theme switcher implemented (switches OS themes)

dhdemerson committed:
Get rid of tab separator from horizontal tabs

dhdemerson committed:
Dark theme first pass

dhdemerson committed:
Add chameleon hover color

dhdemerson committed:
Don't move label on selection

dhdemerson committed:
Background for favicons

dhdemerson committed:
first pass light theme

dhdemerson committed:
Get rid of tabs margin on light and dark themes

dhdemerson committed:
Vertically center close tab button

dhdemerson committed:
Added theme files to makefile

dhdemerson committed:
Seamless enabling/disabling, toolbar always visible

dhdemerson committed:
Merge branch 'no-restart' into themes

Commit:
bump min-width to 20px

Commit:
doubled width, made splitter more hittable

Commit:
adjustments on vertical splitter

Commit:
little more cleanup in osx.css

Commit:
update README with install link

darrinhenein committed:
Merge pull request #3 from robcee/master

Minimum Width, Larger Hit Size on Sidebar Splitter

dhdemerson committed:
initial light theme tab toolbar styling

dhdemerson committed:
No restart to switch themes

dhdemerson committed:
Make default preference use OS styles

dhdemerson committed:
Don't brighten favicons/backgrounds on hover

dhdemerson committed:
Updated XPI

dhdemerson committed:
Fixed broken styles for tab toolbar on windows

darrinhenein committed:
Merge resolution

darrinhenein committed:
Merge branch 'dhdemerson-themes'

darrinhenein committed:
Added xpi

Commit:
Rename to "Vertical Tabs (Simplified)".

Strip group and multiselect support.

Commit:
Move to MPL2, since we're MPL1 tri-licensed already.

Also remove a hardcoded link from the README.

Commit:
Update the XPI.

ntninja committed:
Add option to hide tab bar in fullscreen mode
